src/components/legend/draw.js

```diff
@@ -0,0 +1,448 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var d3 = require('d3');
+
+var Plotly = require('../../plotly');
+var Lib = require('../../lib');
+var Plots = require('../../plots/plots');
+var Fx = require('../../plots/cartesian/graph_interact');
+var Drawing = require('../drawing');
+var Color = require('../color');
+
+var constants = require('./constants');
+var getLegendData = require('./get_legend_data');
+var style = require('./style');
+var helpers = require('./helpers');
+var anchorUtils = require('./anchor_utils');
+
+
+module.exports = function draw(gd) {
+    var fullLayout = gd._fullLayout;
+
+    if(!fullLayout._infolayer || !gd.calcdata) return;
+
+    var opts = fullLayout.legend,
+        legendData = fullLayout.showlegend && getLegendData(gd.calcdata, opts),
+        hiddenSlices = fullLayout.hiddenlabels || [];
+
+    if(!fullLayout.showlegend || !legendData.length) {
+        fullLayout._infolayer.selectAll('.legend').remove();
+        Plots.autoMargin(gd, 'legend');
+        return;
+    }
+
+    if(typeof gd.firstRender === 'undefined') gd.firstRender = true;
+    else if(gd.firstRender) gd.firstRender = false;
+
+    var legendsvg = fullLayout._infolayer.selectAll('svg.legend')
+        .data([0]);
+    legendsvg.enter().append('svg')
```

Comment:
Can we replace this with a `g` without issue and kill two birds with one stone?

Not sure whether there's any `svg` tag specific attributes being utilized on the top level `svg.legend` element.